src/memberPublicKey.spec.ts

```diff
@@ -0,0 +1,211 @@
+import { getModelForClass, type ReturnModelType } from '@typegoose/typegoose';
+import type { Connection } from 'mongoose';
+
+import { setUpTestDbConnection } from './testUtils/db.js';
+import { makeMockLogging, type MockLogging, partialPinoLog } from './testUtils/logging.js';
+import {
+  MEMBER_MONGO_ID,
+  MEMBER_PUBLIC_KEY_MONGO_ID as PUBLIC_KEY_ID,
+  TEST_OID,
+} from './testUtils/stubs.js';
+import type { ServiceOptions } from './serviceTypes.js';
+import { requireFailureResult, requireSuccessfulResult } from './testUtils/result.js';
+import { getPromiseRejection } from './testUtils/jest.js';
+import { MemberPublicKeyModelSchema } from './models/MemberPublicKey.model.js';
+import { generatePublicKey } from './testUtils/publicKeyGenerator.js';
+import {
+  createMemberPublicKey,
+  deleteMemberPublicKey,
+  getMemberPublicKey,
+} from './memberPublicKey.js';
+import { MemberPublicKeyProblemType } from './MemberPublicKeyProblemType.js';
+
+const publicKey = await generatePublicKey();
+
+describe('member public key', () => {
+  const getConnection = setUpTestDbConnection();
+
+  let mockLogging: MockLogging;
+  let connection: Connection;
+  let serviceOptions: ServiceOptions;
+  let memberPublicKeyModel: ReturnModelType<typeof MemberPublicKeyModelSchema>;
+  beforeEach(() => {
+    mockLogging = makeMockLogging();
+    connection = getConnection();
+    serviceOptions = {
+      dbConnection: connection,
+      logger: mockLogging.logger,
+    };
+    memberPublicKeyModel = getModelForClass(MemberPublicKeyModelSchema, {
+      existingConnection: connection,
+    });
+  });
+
+  describe('createMemberPublicKey', () => {
+    test('Member public key should be created', async () => {
+      const memberPublicKey = await createMemberPublicKey(
+        MEMBER_MONGO_ID,
+        {
+          publicKey: publicKey.toString('base64'),
```

Comment:
We're doing this in a few tests, so I think we should just have a constant for it.